app/src/test/kotlin/fr/free/nrw/commons/nearby/NearbyParentFragmentPresenterTest.kt

```diff
@@ -0,0 +1,320 @@
+package fr.free.nrw.commons.nearby
+
+import com.nhaarman.mockitokotlin2.*
+import fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsDao
+import fr.free.nrw.commons.location.LatLng
+import fr.free.nrw.commons.location.LocationServiceManager
+import fr.free.nrw.commons.nearby.contract.NearbyParentFragmentContract
+import fr.free.nrw.commons.nearby.presenter.NearbyParentFragmentPresenter
+import org.junit.Before
+import org.junit.Test
+import org.junit.Assert.*
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+
+
+/**
+ * The unit test class for NearbyParentFragmentPresenter
+ */
+class NearbyParentFragmentPresenterTest {
+    @Mock
+    internal lateinit var nearbyParentFragmentView: NearbyParentFragmentContract.View
+    @Mock
+    internal lateinit var bookmarkLocationsDao: BookmarkLocationsDao
+    @Mock
+    internal lateinit var latestLocation: LatLng
+    @Mock
+    internal lateinit var cameraTarget: LatLng
+    @Mock
+    internal lateinit var selectedLabels: List<Label>
+
+    private lateinit var nearbyPresenter: NearbyParentFragmentPresenter
+    private lateinit var latestLocationSpy: LatLng
+    private lateinit var mapboxCameraTarget: com.mapbox.mapboxsdk.geometry.LatLng
+
+    /**
+     * initial setup
+     */
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        nearbyPresenter = Mockito.spy(NearbyParentFragmentPresenter(bookmarkLocationsDao))
+        nearbyPresenter.attachView(nearbyParentFragmentView)
+    }
+
+    /**
+     * Tests nearby operations are initialized
+     */
+    @Test
+    fun testInitializeNearbyMapOperations() {
+        nearbyPresenter.initializeMapOperations()
+        verify(nearbyParentFragmentView).addSearchThisAreaButtonAction()
+        verify(nearbyParentFragmentView).setCheckBoxAction()
```

Comment:
no verification for 
```
        lockUnlockNearby(false);
        updateMapAndList(LOCATION_SIGNIFICANTLY_CHANGED);
```

Reply:
and `initializeMapOperations` should be a private function, only invoked by `onMapReady`

Reply:
> no verification for
> 
> ```
>         lockUnlockNearby(false);
>         updateMapAndList(LOCATION_SIGNIFICANTLY_CHANGED);
> ```
Adding them